core/services/workflows/syncer/handler.go

```diff
@@ -135,12 +158,119 @@ func (h *eventHandler) Handle(ctx context.Context, event WorkflowRegistryEvent)
 }
 
 // workflowRegisteredEvent handles the WorkflowRegisteredEvent event type.
-// TODO: Implement this method
 func (h *eventHandler) workflowRegisteredEvent(
-	_ context.Context,
-	_ WorkflowRegistryEvent,
+	ctx context.Context,
+	event WorkflowRegistryEvent,
 ) error {
-	return ErrNotImplemented
+	payload, ok := event.Data.(WorkflowRegistryWorkflowRegisteredV1)
```

Comment:
Nit: move this up into whatever calls the handler and make the handler event type specific; it'll clutter this method less, and you won't need a guard clause in every handler